### PR TITLE
EVG-19681 Ensure invalid regular expressions aren't applied to logs and introduce additional validation

### PR DESCRIPTION
**Pull request by khelif96**
EVG-19681

### Description 
It looks like this bug was caused by some users accessing a broken parsley link which may have had a broken regular expression in the URL. Since the regular expression was invalid it would throw an error when we tried to build it in the `constructRegexToMatch` function. 
* This PR adds some error handling and reporting to `constructRegexToMatch`.
* Also updates the side nav to show the errored regular expression and provides a path for users to fix it. 

### Screenshots
image
image

https://github.com/evergreen-ci/parsley/assets/4605522/491b0b21-7394-40bb-be9b-459f6bf4fac5

### Testing 
Replicated the error locally and wrote some tests to fix it.

### CODE REVIEW

File: src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx [hunk 1]
```diff
@@ -40,11 +41,27 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
...
+        setRegexpError((e as Error).message);
+      }
+    }
+  }, [name]);
 
   const resetEditState = () => {
     setIsEditing(false);
     setNewFilterName(name);
-    setIsValid(true);
+    // setIsValid(true);
```
Comment (sophstad) [inline src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx:64]:
```suggestion
```

File: src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx [hunk 2]
```diff
@@ -62,7 +79,16 @@ const FilterGroup: React.FC<FilterGroupProps> = ({
   return (
     <FilterContainer data-cy={dataCy}>
       <FilterHeader>
-        <Badge>FILTER</Badge>
+        <BadgeWrapper>
+          <Badge>FILTER</Badge>
```
Comment (sophstad) [inline src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx:83]: [nit] The badge applies capitalization itself
```suggestion
          <Badge>Filter</Badge>
```

**Review by sophstad — CHANGES_REQUESTED** [review]

**Comment by khelif96** [comment]
Did some refactoring to surface the errors regardless of the scenario.

File: src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx (hunk 2 above)
Comment (sophstad) [reply src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx:72]: Bump

File: src/utils/validators/index.ts [hunk 3]
```diff
@@ -12,4 +12,24 @@ const validateRegexp = (regexp: string): boolean => {
...
+ * @param regexp - the regexp to test
+ * @returns
+ */
+const getRegexpError = (regexp: string): string => {
+  try {
+    // eslint-disable-next-line no-new
+    new RegExp(regexp);
+    return "";
+  } catch (e: any) {
+    console.log(e.message);
```
Comment (sophstad) [inline src/utils/validators/index.ts:26]:
```suggestion
```

File: src/utils/validators/index.ts [hunk 4]
```diff
@@ -12,4 +12,24 @@ const validateRegexp = (regexp: string): boolean => {
...
+    const message = e.message.replace(
```
Comment (sophstad) [inline src/utils/validators/index.ts:27]: [nit] Could be helpful to mention we're removing this string from the beginning of the error message

File: src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx (hunk 2 above)
Comment (sophstad) [reply src/components/SidePanel/NavGroup/FilterNavGroup/FilterGroup/index.tsx:72]: Bump

**Review by sophstad — APPROVED** [review]
A few small notes but LGTM!

**Pull request merged**